pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -272,6 +289,16 @@ func managePod(client coreclientv1.ConfigMapsGetter, recorder events.Recorder, o
 	}
 	required.Spec.Containers[0].Args[0] = strings.ReplaceAll(containerArgsWithLoglevel[0], "${VERBOSITY}", verbosity)
 
+	// apply graceful termination timeouts
+	containerArgsWithGracefulTerminationDuration := required.Spec.Containers[0].Args
+	if argsCount := len(containerArgsWithGracefulTerminationDuration); argsCount > 1 {
+		return nil, false, fmt.Errorf("expected only one container argument for setting graceful termination duration, got %d", argsCount)
+	}
+	watchTerminationDuration, podTerminationGracePeriodSeconds := gracefulTerminationDurations(infrastructureLister)
+	// we don't need special formatting since the only thing we are changing is the graceful termination duration
+	required.Spec.Containers[0].Args[0] = strings.ReplaceAll(containerArgsWithLoglevel[0], "${GRACEFUL_TERMINATION_DURATION}", fmt.Sprintf("%ds", watchTerminationDuration))
```

Comment:
I am not a fan of these replacements deep in yaml structs. Can't we do this in a more generic way, maybe even on the whole manifest text?

Reply:
Same applies to the verbosity above.

Reply:
we can't apply it to the entire manifest because some variables might require special escaping because they are used as arguments in a bash script.

Reply:
which exactly? `${VERBOSITY}` is a number. `${GRACEFUL_TERMINATION_DURATION}` is a number. Neither needs escaping.